app/include/bfx-api-cpp/TRequest.hpp

```diff
@@ -0,0 +1,288 @@
+        }
+        return response;
+      };
+
+      string post(string inPath, string json = "") {
+        response = "";
+        if (curlPOST) {
+          path = inPath;
+          string url = endpoint + path;
+          string payload;
+          getBase64(json, payload);
+
+          if (accessKey != "") {
+            header["X-BFX-APIKEY"] = accessKey;
+          }
+
+          if (secretKey != "") {
+            header["X-BFX-SIGNATURE"] = getSignature(payload);
+          }
+
+          if (payload != "") {
+           header["X-BFX-PAYLOAD"] = payload;
+          }
+
+          setupHeader();
+          curl_easy_setopt(curlPOST, CURLOPT_HTTPHEADER, curlHeader);
+          curl_easy_setopt(curlPOST, CURLOPT_POST, 1);
+          curl_easy_setopt(curlPOST, CURLOPT_POSTFIELDS, "\n");
+          curl_easy_setopt(curlPOST, CURLOPT_TIMEOUT, CURL_TIMEOUT);
+          curl_easy_setopt(curlPOST, CURLOPT_URL, url.c_str());
+          curl_easy_setopt(curlPOST, CURLOPT_VERBOSE, CURL_DEBUG_VERBOSE);
+          curl_easy_setopt(curlPOST, CURLOPT_WRITEDATA, &response);
+          curl_easy_setopt(curlPOST, CURLOPT_WRITEFUNCTION, writeCallback);
+
+          curlStatusCode = curl_easy_perform(curlPOST);
+          clearHeader();
+          // libcurl internal error handling
+          if (curlStatusCode != CURLE_OK) {
+            cerr << "libcurl error in Request.post():" << endl;
+            cerr << "CURLcode: " << curlStatusCode << endl;
+          }
+        } else {
+          cerr << "curl not properly initialized curlPOST = nullptr";
+        }
+
+        return response;
+      };
+
+      string parseParams(map<string, string> params) {
+        string pp = "";
+        map<string, string>::iterator it;
+        for (it = params.begin(); it != params.end(); it++) {
+          pp += it->first + "=" + it->second + "&";
+        }
+        return pp;
+      };
+
+      string getSignature(string payload) {
+        string signature;
+        getHmacSha384(secretKey, payload, signature);
+        return signature;
+      }
+
+      const CURLcode getLastStatusCode() const noexcept {
+        return curlStatusCode;
+      }
+
+      const string getLastResponse() const noexcept {
+        return response;
+      }
+
+      const string getLastPath() const noexcept {
+        return path;
+      }
+
+      const bool hasError() const noexcept {
+        return curlStatusCode != CURLE_OK;
+      }
+
+      void setSecretKey(string inSecretKey) {
+        secretKey = inSecretKey;
+      }
+
+      void setAccessKey(string inAccessKey) {
+        accessKey = inAccessKey;
+      }
+
+      void setHeader(map<string, string> inHeader) {
+        header = inHeader;
+      }
+
+    private:
+      ////////////////////////////////////////////////////////////////////////
+      // Private properties
+      ////////////////////////////////////////////////////////////////////////
+      string endpoint = "";
```

Comment:
I think string initialization `string endpoint = "";` is unnecessary. Simple less verbose `string endpoint;` should do the same.

> (1) empty string constructor (default constructor)
    Constructs an empty string, with a length of zero characters.